### PR TITLE
Add test to ensure duplicate findings are deleted in the proper order

### PR DESCRIPTION
**Pull request by Jino-T**
Add test to ensure that when a maximum duplicate value is surpassed, the oldest finding (based on date attribute on finding) is deleted. We want to avoid duplicate findings being deleted in order of finding id.

[sc-12742]

### CODE REVIEW

File: unittests/test_duplication_loops.py [hunk 1]
```diff
@@ -374,6 +375,38 @@ def test_list_relations_for_three_reverse(self):
...
+        self.assertFalse(
+            Finding.objects.filter(id=self.finding_c.id).exists(),
+            "The oldest duplicate (Finding c) should have been deleted.",
+        )
+        # Finding b (2024) should exist
+        self.assertTrue(
+            Finding.objects.filter(id=self.finding_b.id).exists(),
+            "The newest duplicate (Finding B) should still exist.",
+        )
+        # Finding A should now only have 2 duplicate in its set
```
Comment (Maffooch) [inline unittests/test_duplication_loops.py]:
```suggestion
        # Finding A should now only have 1 duplicate in its set
```

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved
